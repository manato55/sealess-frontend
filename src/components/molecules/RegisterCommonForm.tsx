import React,{Dispatch, SetStateAction} from 'react'
import {useAuth} from '../../hooks/useAuth'
import Input from '../atoms/Input'
import ErrorMessageWrapper from '../atoms/ErrorMessageWrapper'
import { useRecoilValue } from 'recoil'
import { authErrorMessage, eachErrorFlag } from '../../store/atom'



type Props = {
    setEmail: Dispatch<SetStateAction<string>>
    setPassword: Dispatch<SetStateAction<string>>
    email: string,
    password: string,
}

export const RegisterCommonForm = (props: Props): React.ReactElement  => {
    const errorMessage = useRecoilValue(authErrorMessage)
    const errorFlag = useRecoilValue(eachErrorFlag)
    


    const emailHandler = (e: React.ChangeEvent<HTMLInputElement>):void => {
        props.setEmail(e.target.value)
    }

    const passwordHandler = (e: React.ChangeEvent<HTMLInputElement>):void => {
        props.setPassword(e.target.value)
    }

    return (
        <div>
            <ErrorMessageWrapper>{errorFlag.email && errorMessage.email[0]}</ErrorMessageWrapper>
            <Input
                type="email"
                value={props.email}
                onChange={(e) => emailHandler(e)}
                placeholder="メールアドレス"
            />
            <ErrorMessageWrapper>{errorFlag.password && errorMessage.password[0]}</ErrorMessageWrapper>
            <Input
                type="password"
                value={props.password}
                onChange={(e) => passwordHandler(e)}
                placeholder="パスワード"
            />
        </div>
    )
}



export default RegisterCommonForm