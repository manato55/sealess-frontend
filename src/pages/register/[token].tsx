import RegisterToken from '../../components/organisms/RegisterToken';

const Token = () => {
  return (
    <>
      <RegisterToken />
    </>
  );
};

export default Token;
