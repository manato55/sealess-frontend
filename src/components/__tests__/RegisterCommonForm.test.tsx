import React from 'react';
import RegisterCommonForm from '../molecules/RegisterCommonForm';
import { mount, shallow } from 'enzyme';
import 'jsdom-global/register';
import { RecoilRoot } from 'recoil';

describe('RegisterCommonForm test', () => {
  let container;
  const onChangeMockEmail = jest.fn();
  const onChangeMockPassword = jest.fn();

  beforeEach(() => {
    container = mount(
      <RecoilRoot>
        <RegisterCommonForm
          setEmail={onChangeMockEmail}
          setPassword={onChangeMockPassword}
          email={'test@test.com'}
          password={'testtest'}
        />
      </RecoilRoot>
    );
  });

  test('email input test', () => {
    container.find('input[type="email"]').simulate('change');
    expect(onChangeMockEmail).toHaveBeenCalledWith('test@test.com');
  });

  test('password input test', () => {
    container.find('input[type="password"]').simulate('change');
    expect(onChangeMockPassword).toHaveBeenCalledWith('testtest');
  });

  test('snapshot', () => {
    expect(container).toMatchSnapshot();
  });
});
