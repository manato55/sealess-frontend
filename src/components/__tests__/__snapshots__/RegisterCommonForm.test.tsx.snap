// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterCommonForm test snapshot 1`] = `
<RecoilRoot>
  <RecoilRoot_INTERNAL>
    <Batcher
      setNotifyBatcherOfChange={[Function]}
    />
    <RegisterCommonForm
      email="test@test.com"
      password="testtest"
      setEmail={
        [MockFunction] {
          "calls": Array [
            Array [
              "test@test.com",
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        }
      }
      setPassword={
        [MockFunction] {
          "calls": Array [
            Array [
              "testtest",
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        }
      }
    >
      <div>
        <styled.p>
          <p
            className="sc-gtsrHT DoTxJ"
          />
        </styled.p>
        <styled.input
          editable={true}
          onChange={[Function]}
          placeholder="メールアドレス"
          type="email"
          value="test@test.com"
          visible={false}
        >
          <input
            className="sc-bdnxRM elxLYl"
            onChange={[Function]}
            placeholder="メールアドレス"
            type="email"
            value="test@test.com"
          />
        </styled.input>
        <styled.p>
          <p
            className="sc-gtsrHT DoTxJ"
          />
        </styled.p>
        <styled.input
          editable={true}
          onChange={[Function]}
          placeholder="パスワード"
          type="password"
          value="testtest"
          visible={false}
        >
          <input
            className="sc-bdnxRM elxLYl"
            onChange={[Function]}
            placeholder="パスワード"
            type="password"
            value="testtest"
          />
        </styled.input>
      </div>
    </RegisterCommonForm>
  </RecoilRoot_INTERNAL>
</RecoilRoot>
`;
